projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/execution/JobExecutionService.java

```diff
@@ -120,6 +111,24 @@ public String startDataJobExecution(
           ExecutionStatus.SUBMITTED,
           startedBy,
           OffsetDateTime.now());
+      try {
+        // Start K8S Job
+        dataJobsKubernetesService.startNewCronJobExecution(
+            jobDeploymentStatus.getCronJobName(),
+            executionId,
+            annotations,
+            envs,
+            extraJobArguments,
+            jobName);
+      } catch (Exception e) {
+        // rollback data job execution
+        jobExecutionRepository.deleteDataJobExecutionByIdAndDataJobAndStatusAndType(
```

Comment:
```suggestion
        jobExecutionRepository.deleteById(executionId);
```
isn't it enough?

Reply:
AFAIK, it would be enough. 
Given that I don't know the codebase too well I wanted to err on the side of caution. 
I only wanted to delete the line if I'm sure no one has changed it since I added it.
WDYT will I go with your suggested change?

Reply:
Ok, leave it as it is.